packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -542,6 +552,26 @@ class CircularProgressIndicator extends ProgressIndicator {
   /// The width of the line used to draw the circle.
   final double strokeWidth;
 
+  /// The shape of the progress indicator's stroke ends.
+  ///
+  /// Determines the appearance of the stroke ends in the progress indicator.
+  /// By default, [strokeCap] is null, causing the stroke ends to be set to
+  /// [StrokeCap.square] when [value] is null (indeterminate), and to
+  /// [StrokeCap.butt] when [value] is not null (determinate).
+  ///
+  /// Setting [strokeCap] to [StrokeCap.round] results in rounded stroke ends.
+  /// With [strokeCap] set to [StrokeCap.butt] and [value] == null, the
+  /// indicator exhibits a different indeterminate animation, disappearing
+  /// completely before reappearing at its minimum value.
+  ///
+  /// Using [strokeCap] as [StrokeCap.square] when [value] is not null leads to
+  /// an unconventional display of [value], starting and ending slightly
+  /// off the expected angles. While not recommended, this behavior can be
```

Comment:
I don't think it is recommended because that's simply not how CircularProgressIndicator was intended to be used (based on the way it dances with butt and square for different scenarios). I think it is fair to warn the users it is probably not what they want (this will suit 0.00001% of them). ShortcutActivator doesn't recommend certain pattern because a different one "closely resambles that of typical platforms". I think we are able to not recommend something that.

 But I can try better wording. What do you think of this?
<img width="695" alt="image" src="https://user-images.githubusercontent.com/351125/230980950-1134dfa6-c0df-4ed2-8789-67cebc9ed4e0.png">

Reply:
GPT-4 provided:
> Maybe. I tried with GPT-4, it said:
> Be careful when using [StrokeCap.square] with a non-null [value]. The start and end angles may not be what you expect. For example, with a [value] of 0.5, you might expect the start and end angles to be 90 and 270 degrees. However, the indicator could begin at 85 degrees and end at 275 degrees. This doesn't follow the usual design rules, so use it with caution.

Reply:
I liked the lighter, I'll use it. Let me know if you feel it is better.